client/extensions/zoninator/components/settings/zone/index.jsx

```diff
@@ -1,37 +1,93 @@
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
-import { flowRight, noop } from 'lodash';
+import { flowRight } from 'lodash';
+import { FieldArray, FormSection, reduxForm } from 'redux-form';
 
 /**
  * Internal dependencies
  */
+import PostsList from './posts-list';
+import Button from 'components/button';
+import Card from 'components/card';
+import FormButton from 'components/forms/form-button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
 import HeaderCake from 'components/header-cake';
+import ReduxFormTextarea from 'components/redux-forms/redux-form-textarea';
+import ReduxFormTextInput from 'components/redux-forms/redux-form-text-input';
+import SectionHeader from 'components/section-header';
 import { getSelectedSiteSlug } from 'state/ui/selectors';
+import { settingsPath } from '../../../app/util';
 
-const Zone = ( { siteSlug, translate } ) => (
-	<div>
-		<HeaderCake backHref={ `/extensions/zoninator/${ siteSlug }` } onClick={ noop }>
-			{ translate( 'Edit zone' ) }
-		</HeaderCake>
-	</div>
-);
-
-Zone.propTypes = {
-	siteSlug: PropTypes.string,
-};
-
-const connectComponent = connect( state => {
-	return {
-		siteSlug: getSelectedSiteSlug( state ),
-	};
+const form = 'extensions.zoninator.editZone';
+
+class Zone extends Component {
+
+	onFormSubmit = ( event ) => {
+		event.preventDefault();
+	}
+
+	render() {
+		const { siteSlug, translate } = this.props;
+
+		return (
+			<div>
+				<HeaderCake
+					backHref={ `${ settingsPath }/${ siteSlug }` }
+					actionButton={ <Button compact primary scary>{ translate( 'Delete' ) }</Button> } >
+					{ translate( 'Edit zone' ) }
+				</HeaderCake>
+
+				<form onSubmit={ this.onFormSubmit }>
+					<FormSection name="details">
+						<SectionHeader label={ translate( 'Zone label' ) }>
+							<FormButton compact />
+						</SectionHeader>
+						<Card>
+							<FormFieldset>
+								<FormLabel htmlFor="zoneName">{ translate( 'Zone name' ) }</FormLabel>
+								<ReduxFormTextInput name="zoneName" />
+							</FormFieldset>
+
+							<FormFieldset>
+								<FormLabel htmlFor="zoneDescription">{ translate( 'Zone description' ) }</FormLabel>
+								<ReduxFormTextarea name="zoneDescription" />
+							</FormFieldset>
+						</Card>
+					</FormSection>
+
+					<FormSection name="content">
+						<SectionHeader label={ translate( 'Zone content' ) }>
+							<FormButton compact />
+						</SectionHeader>
+						<Card>
+							<FieldArray
+								rerenderOnEveryChange
+								name="posts"
+								component={ PostsList } />
+						</Card>
+					</FormSection>
+				</form>
+			</div>
+		);
+	}
+}
+
+const connectComponent = connect( state => ( {
+	siteSlug: getSelectedSiteSlug( state ),
+} ) );
+
+const createReduxForm = reduxForm( {
+	enableReinitialize: true,
+	form,
 } );
 
 export default flowRight(
 	connectComponent,
 	localize,
+	createReduxForm,
```

Comment:
I'd put `createReduxForm` before `localize`. I recall faintly that e.g. putting `localize` before `connectComponent` can cause problems with props not being properly passed thru, which is why I usually try to call it last.

Reply:
(I might be overly cautious here. But maybe it's okay as sort of a convention.)